src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -715,33 +704,108 @@ const TransactionBuilderContextProvider = ({
       setTransactionBlockFieldValidationError,
       setCrossChainActionGasToken,
     }),
-    [],
+    []
   );
 
   const hideMenu = () => setShowMenu(false);
 
   const hasTransactionBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.ASSET_BRIDGE
   );
 
   const hasKlimaBlockAdded = transactionBlocks.some(
-    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE,
+    (transactionBlock) => transactionBlock.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE
   );
 
   const crossChainActionsInProcessing = useMemo(() => {
     if (!processingCrossChainActionIds?.length) return;
-    return dispatchedCrossChainActions?.filter(
-      (crossChainAction) => processingCrossChainActionIds.some((id) => id === crossChainAction.id),
+    return dispatchedCrossChainActions?.filter((crossChainAction) =>
+      processingCrossChainActionIds.some((id) => id === crossChainAction.id)
     );
   }, [processingCrossChainActionIds, dispatchedCrossChainActions]);
 
-  const hasProcessingUnsent = useMemo(() => !!getFirstCrossChainActionByStatus(
-    dispatchedCrossChainActions ?? [],
-    CROSS_CHAIN_ACTION_STATUS.UNSENT,
-  ), [dispatchedCrossChainActions]);
+  const hasProcessingUnsent = useMemo(
+    () => !!getFirstCrossChainActionByStatus(dispatchedCrossChainActions ?? [], CROSS_CHAIN_ACTION_STATUS.UNSENT),
+    [dispatchedCrossChainActions]
+  );
 
   const [showMulticallOptions, setShowMulticallOptions] = useState<string | null>(null);
 
+  const onBuyClick = async () => {
+    const maticSdk = getSdkForChainId(CHAIN_ID.POLYGON);
+
+    if (!accountAddress || !maticSdk) return;
+
+    let account = await maticSdk.getAccount();
+    if (!account || account.address !== accountAddress) {
+      await maticSdk.computeContractAccount();
+      account = await maticSdk.getAccount();
+    }
+
+    if (!account) return;
+
+    let base64Hash = '';
+    let onRampOptions = {
+      lang: 'fr',
+      tab: 'buy',
+      tabs: 'buy',
+      chain: 'matic_mainnet',
+      net: 'matic_mainnet',
+      nets: 'arbitrum_mainnet,avalanche_mainnet,bsc_mainnet,fantom_mainnet,mainnet,optimism_mainnet,xdai_mainnet',
+      crys: 'AVAX,BNB,BTCB,BUSD,DAI,ETH,FRAX,LUSD,MAI,MATIC,RBTC,RDOC,RIF,USDC,USDT,WBTC,WETH,XCHF,XDAI,XTZ',
+      rfr: 'etherspot',
+      bsc: 'GBP',
+      bdc: 'MATIC',
+      mode: 'dark',
+      hash: '',
+      code: '',
+      addr: '',
+    };
+    const code = Math.floor(Math.random() * 8999) + 1000;
+    const message = 'MtPelerin-' + code;
+    onRampOptions.code = code.toString();
+    onRampOptions.addr = account.address || '';
+
+    if (account.state === AccountStates.UnDeployed) {
+      const submittedGateway = await deployAccount(maticSdk);
+      if (!submittedGateway) return;
```

Comment:
would be good to inform user about what happened

Reply:
also additionally to get right error maybe add excception handler for `deployAccount`

Reply:
also would be good to inform user while waiting

Reply:
Changed the "Buy" text to show "Deploying...", and added some alerts. Also I've left the error as generic, as I wasn't sure if the error from etherspot is something that we want the user to see